### PR TITLE
impl(docfx): support `programlisting` elements

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10890)

### CODE REVIEW

File: docfx/doxygen2markdown.cc [hunk 1]
```diff
@@ -327,7 +327,9 @@ bool AppendIfDocTitleCmdGroup(std::ostream& os, MarkdownContext const& ctx,
 bool AppendIfDocCmdGroup(std::ostream& os, MarkdownContext const& ctx,
                          pugi::xml_node const& node) {
   if (AppendIfDocTitleCmdGroup(os, ctx, node)) return true;
-  // Unexpected: hruler, preformatted, programlisting, verbatim, indexentry
+  // Unexpected: hruler, preformatted
+  if (AppendIfProgramListing(os, ctx, node)) return true;
+  // Uexpected: verbatim, indexentry
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:332]: s/Uexp/Unexp/

File: docfx/doxygen2markdown.cc [hunk 2]
```diff
@@ -361,6 +363,115 @@ bool AppendIfParagraph(std::ostream& os, MarkdownContext const& ctx,
...
+//       <xsd:element name="codeline" type="codelineType" minOccurs="0" maxOccurs="unbounded" />
+//     </xsd:sequence>
+//     <xsd:attribute name="filename" type="xsd:string" use="optional"/>
+//   </xsd:complexType>
+// clang-format on
+bool AppendIfProgramListing(std::ostream& os, MarkdownContext const& ctx,
+                            pugi::xml_node const& node) {
+  if (std::string_view{node.name()} != "programlisting") return false;
+  // Start with a new paragraph, with the right level of indentation, and a new
+  // code fence:
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:381]: s/:/./

File: docfx/doxygen2markdown.cc [hunk 3]
```diff
@@ -361,6 +363,115 @@ bool AppendIfParagraph(std::ostream& os, MarkdownContext const& ctx,
...
+    if (AppendIfPlainText(os, ctx, child)) continue;
+    if (AppendIfHighlightSp(os, ctx, child)) continue;
+    if (AppendIfHighlightRef(os, ctx, child)) continue;
+    UnknownChildType(__func__, child);
+  }
+  return true;
+}
+
+// A `<sp>` element is just a space. It seems that Doxygen does not use the
+// `value` attribute, we will leave that unhandled.
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:440]: s/attribute, we/attribute, so we/

File: docfx/doxygen2markdown.cc [hunk 4]
```diff
@@ -361,6 +363,115 @@ bool AppendIfParagraph(std::ostream& os, MarkdownContext const& ctx,
...
+  if (std::string_view{node.name()} != "sp") return false;
+  // Leave the 'value' attribute unhandled. It is probably the number of spaces,
+  // but without documentation it is hard to say. Since it is unused, this
+  // approach seems safer.
+  if (!std::string_view{node.attribute("value").name()}.empty()) return false;
+  os << ' ';
+  return true;
+}
+
+// A `ref` element inside a `highlight` element has `refTextType`. Which is
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:456]: s/. Which/, which/

**Review by devbww — APPROVED** [review]

File: docfx/doxygen2markdown.cc (hunk 1 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed

File: docfx/doxygen2markdown.cc (hunk 2 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed.

File: docfx/doxygen2markdown.cc (hunk 3 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed.

File: docfx/doxygen2markdown.cc (hunk 4 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed

**Pull request merged**